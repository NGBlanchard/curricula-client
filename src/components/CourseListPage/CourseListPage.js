import React, { Component } from 'react'
import CourseContext from '../../context/CourseContext'
import CourseApiService from '../../services/course-api-service'
import { Section } from '../Utils/Utils'
import CourseListItem from '../CourseListItem/CourseListItem'
import Search from '../Search/Search';

export default class CourseListPage extends Component {
  
  static contextType = CourseContext

  componentDidMount() {
    this.context.clearError()
    CourseApiService.getCourses()
      .then(this.context.setCourseList)
      .catch(this.context.setError)
  }
  
  renderCourses() {
    const { courseList = [] } = this.context
    return courseList.map(course =>
      <CourseListItem
        key={course.id}
        course={course}
      />
    )
  }

  render() {
    const { error } = this.context
    return (
      <>
      <Search />
      <Section list className='CourseListPage'>
        {error
          ? <p className='red'>There was an error. try something else, y'know?</p>
          : this.renderCourses()}
      </Section>
      </>
    )
  }
}
