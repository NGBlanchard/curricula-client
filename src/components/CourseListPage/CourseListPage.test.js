import CourseListPage from "./CourseListPage";
import { BrowserRouter as Router } from "react-router-dom";
import React from "react";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<CourseListPage />", () => {
  const course = {
    id: 4,
    title: "title",
    description: "description",
    readings: "readings",
    notes: "notes",
    topic: "topic",
    date_created: "date",
    author_id: 2
  };

  const context = {
    clearError() {
      return;
    },
    setCourseList() {
      return;
    },
    filterTopics() {
      const filteredtopics = this.state.courseList.map(t => t.topic);
      this.setState({
        topics: filteredtopics
      });
    },
    courseList: [],
    filteredList: []
  };
  it("Renders without crashing", () => {
    mount(
      <Router>
        <CourseListPage key={course.id} course={course} />
      </Router>,
      context
    );
  });
  it("renders the page by default", () => {
    const wrapper = mount(
      <Router>
        <CourseListPage key={course.id} course={course} />
      </Router>,
      context
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
