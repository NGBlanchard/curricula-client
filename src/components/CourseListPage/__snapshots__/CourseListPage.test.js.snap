// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseListPage /> renders the page by default 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CourseListPage
      course={
        Object {
          "author_id": 2,
          "date_created": "date",
          "description": "description",
          "id": 4,
          "notes": "notes",
          "readings": "readings",
          "title": "title",
          "topic": "topic",
        }
      }
      key="4"
    >
      <Nav>
        <nav
          className="Nav"
        >
          <img
            alt="logo"
            className="logo"
            src="../../favicon.ico"
          />
          <h1>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
            >
              <Route
                path="\\\\/"
              >
                <Link
                  aria-current="page"
                  className="active"
                  replace={false}
                  style={Object {}}
                  to="/"
                >
                  <a
                    aria-current="page"
                    className="active"
                    href="/"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Curricula
                  </a>
                </Link>
              </Route>
            </NavLink>
          </h1>
          <Bull>
            <span
              className="Bull"
            >
               • 
            </span>
          </Bull>
          <Link
            replace={false}
            to="/create"
          >
            <a
              href="/create"
              onClick={[Function]}
            >
              Create
            </a>
          </Link>
          <Bull>
            <span
              className="Bull"
            >
               • 
            </span>
          </Bull>
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/profile"
          >
            <Route
              path="\\\\/profile"
            >
              <Link
                aria-current={null}
                replace={false}
                to="/profile"
              >
                <a
                  aria-current={null}
                  href="/profile"
                  onClick={[Function]}
                >
                  Profile
                </a>
              </Link>
            </Route>
          </NavLink>
          <Bull>
            <span
              className="Bull"
            >
               • 
            </span>
          </Bull>
          <div
            className="Nav__not-logged-in"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/register"
            >
              <Route
                path="\\\\/register"
              >
                <Link
                  aria-current={null}
                  replace={false}
                  to="/register"
                >
                  <a
                    aria-current={null}
                    href="/register"
                    onClick={[Function]}
                  >
                    Register
                  </a>
                </Link>
              </Route>
            </NavLink>
            <Bull>
              <span
                className="Bull"
              >
                 • 
              </span>
            </Bull>
            <Link
              replace={false}
              to="/login"
            >
              <a
                href="/login"
                onClick={[Function]}
              >
                Log in
              </a>
            </Link>
          </div>
        </nav>
      </Nav>
      <Search
        courseList={Array []}
        filterList={[Function]}
      >
        <div
          className="topic-sort-container"
        >
          <p>
            Sort by Course Topic: 
          </p>
          <select
            className="topic-sort"
            onChange={[Function]}
          >
            <option
              className="deselect-topic"
              key="none"
              value=""
            >
              None
            </option>
          </select>
        </div>
      </Search>
      <Section
        className="CourseListPage"
        list={true}
      >
        <section
          className="Section Section--list CourseListPage"
        />
      </Section>
    </CourseListPage>
  </Router>
</BrowserRouter>
`;
